classes/class2/README.md

```diff
@@ -1 +1,519 @@
+5.678912345678913
+```
+
+The `Number()` typecast supports strings and floating point numbers (and even rounds!). No need to write your own number parsing code!
+
+The next behavior is arguably worse than throwing an error, but also cool: You *can* do string multiplication in JavaScript!
+
+```node
+> '3' * '4'
+12
+```
+
+Not really, though. It's a number now.
+
+```node
+> typeof('3' * '4')
+'number'
+```
+
+String addition works as expected.
+
+```node
+> 'a' + 'b'
+'ab'
+> '5' + '5'
+'55'
+> '5' - '5'
+0
+```
+
+Well, anyway...
+
+JavaScript, like Python, is dynamically typed. This means that a variable's type isn't determined until runtime. But since JavaScript is also very weakly typed (while Python is not), functions will usually be able to operate on arguments of the wrong type (by coercing them, for example), which means a program can fail unexpectedly long after a type-related mistake is made. In this case, error messages are usually not helpful, so the best thing to do is be careful and think ahead as you write your code.
+
+## Functions
+
+Functions make complexity manageable! JavaScript is all about functions. Let's make some.
+
+At this point we should start putting our code in files. If you have a JavaScript file:
+
+###### functions.js
+```node
+function func1() {
+	return 1;
+}
+```
+
+You can run it like this:
+
+```bash
+$ node functions.js
+```
+
+In `functions.js`, we declare a function named `func1` that takes no arguments and returns the number `1`.
+
+We can add to `functions.js` to learn more about `func1`.
+
+###### functions.js
+```node
+...
+console.log(func1)
+console.log(func1())
+```
+
+The function `console.log` does what you might expect — it logs arguments to the console.
+
+``` bash
+$ node functions.js
+[Function: func1]
+1
+```
+
+The two print statements we added to `functions.js` look almost the same, but the console output shows that they're very different. The first statement prints the function object `func1` — that is, the actual *entity* that represents the function itself (the code, the arguments, the scope in which it was declared — more on scope later). Node says that `func1` is a `Function` named `func1` — nothing unexpected here.
+
+Putting `()` next to the name of a function calls the function, so when we print `func1()` we get the return value of `func1`, which is `1`.
+
+`()` will actually always attempt to "call" the expression to its left, treating it like a function. So you can get errors like this:
+
+```node
+> 5()
+TypeError: number is not a function
+```
+
+A variable can also store a function declaration, so we could have done this instead:
+
+```node
+var func1 = function() {
+	return 1;
+};
+```
+
+The above example wouldn't have acted *quite* the same, though, which brings us to what makes JavaScript functions so special.
+
+### Functions are Objects
+
+The print statements above hinted at this: in JavaScript, functions are objects. So something like this is possible:
```

Comment:
Could you specify that the ability to store functions as variables and pass them around is what demonstrates that they are objects? Otherwise I don't think this is exactly clear.